lib/orderbook/TradingPair.ts

```diff
@@ -267,6 +267,18 @@ class TradingPair {
     return maps.buy.get(orderId) || maps.sell.get(orderId);
   }
 
+  public addOrderHold = (orderId: string, holdAmount: number) => {
+    const order = this.getOwnOrder(orderId);
+    assert((order.hold || 0) + holdAmount <= order.quantity, 'the amount of an order on hold cannot exceed the available quantity');
+    order.hold = order.hold ? order.hold + holdAmount : holdAmount;
```

Comment:
nothing serious, but you're checking twice if `order.hold` exists, and initializing it anyway. Instead you can just initialize it (`order.hold = order.hold || 0`) in the beginning, and avoid the checks.

Reply:
@offerm suggested [here](https://github.com/ExchangeUnion/xud/pull/627#discussion_r231400551) to make hold initialize to 0 to simplify this logic, which I think is a fine idea, so I'll just go with that approach if that's alright with you.

Reply:
Sure.